packages/govuk-frontend/src/govuk/components/header/header.mjs

```diff
@@ -83,23 +83,45 @@ export class Header extends GOVUKFrontendComponent {
     this.$menu = $menu
     this.$menuButton = $menuButton
 
-    // Set the matchMedia to the govuk-frontend desktop breakpoint
-    this.mql = window.matchMedia('(min-width: 48.0625em)')
+    this.setupResponsiveChecks()
+
+    this.$menuButton.addEventListener('click', () =>
+      this.handleMenuButtonClick()
+    )
+  }
+
+  /**
+   * Setup viewport resize check
+   *
+   * @private
+   */
+  setupResponsiveChecks() {
+    const breakpointProperty = '--govuk-frontend-breakpoint-desktop'
+    const breakpointValue = getComputedStyle(document.body).getPropertyValue(
+      breakpointProperty
+    )
+
+    if (!breakpointValue) {
+      throw new ElementError({
+        componentName: 'Header',
+        identifier: `CSS custom property (\`${breakpointProperty}\`)`
```

Comment:
**issue** Thinking the message may lead to people looking to add a custom property on the root of the component's markup, while we look it up on the body 😊 

```suggestion
        identifier: `CSS custom property (\`${breakpointProperty}\`) on document's \`body\``
```

Reply:
How's this? Just pushed it up in 21a519d4fe484ce1b9acb8dc6563b8c165c9e46e

```console
ElementError: Tabs: CSS custom property (`--govuk-frontend-breakpoint-tablet`) on pseudo-class :root not found
```